### ISSUE TITLE
cmake error minimal-ws-server

### ISSUE THREAD

**Issue by nodi-andy** [warmcat/libwebsockets#1378]
hi,
i create cloud debian instance
install all these things: sudo apt-get install git cmake libssl-dev build-essential
i clone the current version : git clone https://github.com/warmcat/libwebsockets
go here: cd libwebsockets/minimal-examples/ws-server/minimal-ws-server
create build : mkdir build
go there: cd build
call cmake: cmake ..

CMake Error at CMakeLists.txt:56 (message):
  This project requires lws must have been configured with LWS_ROLE_WS

funny is, that i solved the problem a couple of months ago but now dont know how.

any help?
thanks
Andy

**Comment by lws-team** [comment]
It's saying that if you want to build that minimal example, since it uses ws your libwebsockets must have been built with websockets support.  Since v3.0 role stuff was introduced, you can turn off most individual role support for build, including websocket support if you just use the library for http things, so if you are on constrained targets you can just get the smallest lws with the stuff you need.

It looks like you configured lws itself (not the minimal example...) with `cmake .. -DLWS_ROLE_WS=0`.  Rebuild lws after `cmake .. -DLWS_ROLE_WS=1` and then try building the example again should solve it.

**Comment by nodi-andy** [comment]
i setted up a new linux instance and did not change any code, also not CMakeList.txt. Recalling cmake with -DLWS_ROLE_WS=1 didnt changed anything. The variable ist defined righ in CMakeList.txt ("on" instead of "1", but should be the same). Also it compiles bin/libwebsockets-test-server during lws compile. But non of minimal examples compiles directly. I remember that i had to change anything in CMakeList but dont know anymore.

**Comment by nodi-andy** [comment]
i also remember now about installing lws and calling ldconfig or similar. Do i have to target the compiled lws bins?

**Comment by lws-team** [comment]
Yes LWS_ROLE_WS is on by default.  But the error you are getting is specifically complaining it can't see that is enabled.

If you didn't disable it, what's in /usr/local/include/lws_config.h?  This is installed with `sudo make install` after lws build.

It should have

```
...
#define LWS_ROLE_H1
...
```

around line 11.

**Comment by nodi-andy** [comment]
oh no!! i didnt make'd with "sudo" and now after "sudo" make install its working. Forgetting "sudo" costs me ca. 2% of my life.
